dmlproc/dmlprocessor.cpp

```diff
@@ -374,9 +374,27 @@ int PackageHandler::synchTableAccess()
 		// tableOidQueue here is the queue holding the waitng transactions for this fTableOid
 		while (true)
 		{
-			tableOidCond.wait(lock);
+            // Log something that we're waiting
+            LoggingID logid(21, fSessionID, fTxnid);
+            logging::Message::Args args1;
+            logging::Message msg(1);
+            ostringstream oss;
+            oss << "Txn is waiting for" << tableOidQueue.front() << " " << dmlPackage->get_SQLStatement() << "; |" << dmlPackage->get_SchemaName() <<"|";
+            args1.add(oss.str());
+            args1.add((uint64_t)fTableOid);
+            msg.format(args1);
+            logging::Logger logger(logid.fSubsysID);
+            logger.logMessage(LOG_TYPE_DEBUG, msg, logid);
```

Comment:
Why should CS log this w/o checking whether debug is turned on or not?

Reply:
The reason behind this block:
What happened is that, if a query has to wait in this queue, the wait happens before the "start sql" log. So we never saw in the log that the query had been submitted until it was released from the queue. And when it was we had no idea it had been queued. With this log, we have a better post-mortem diagnostic. If it gets where this log line spews, then something is most likely broken and that spewing is a clue.